eng/tools/ChangeValidation/ExclusionFileValidation.cs

```diff
@@ -0,0 +1,180 @@
+            .ToList();
+
+        if (excludedFilesInPr.Any())
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.AppendLine($"This PR modifies or creates {excludedFilesInPr.Count} file(s) that are in the exclusion list.");
+            sb.AppendLine("If these files are not needed in the VMR, please delete them in this PR.");
+            foreach (var file in excludedFilesInPr)
+            {
+                sb.AppendLine($" - {file}");
+            }
+            string warningMessage = sb.ToString();
+            AddProcessingMessage(messages, Error(warningMessage));
+        }
+
+        if (filesMatchingNewExclusionRules.Any())
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.AppendLine($"This PR modifies exclusion rules in {SourceMappingsPath}, leading to {filesMatchingNewExclusionRules.Count} newly excluded file(s) in the VMR.");
+            sb.AppendLine("If these files are not needed in the VMR, please delete them in this PR.");
+            foreach (var file in filesMatchingNewExclusionRules)
+            {
+                sb.AppendLine($" - {file}");
+            }
+            string warningMessage = sb.ToString();
+            AddProcessingMessage(messages, Error(warningMessage));
+        }
+
+        if (!excludedFilesInPr.Any() && !filesMatchingNewExclusionRules.Any())
+        {
+            AddProcessingMessage(messages, Success($"Exclusion file validation succeeded."));
+        } else
+        {
+            AddProcessingMessage(messages, Error($"Exclusion file validation failed."));
+        }
+        return messages;
+    }
+
+    private static List<string> GetExclusionPatternsFromBranch(string branchName)
+    {
+        Console.WriteLine($"Retrieving {SourceMappingsPath} from branch `{branchName}`...");
+        string? fileContents = null;
+        try
+        {
+            fileContents = RunGitCommand($"show {branchName}:src/source-mappings.json");
+        } catch (Exception ex)
+        {
+            Console.WriteLine($"Error retrieving {SourceMappingsPath} from branch `{branchName}`: {ex.Message}");
+            return [];
+        }
+
+        if (string.IsNullOrEmpty(fileContents))
+        {
+            Console.WriteLine($"Error: Could not retrieve {SourceMappingsPath} from branch `{branchName}`. The file may not exist or is empty.");
+            return [];
+        }
+
+        JArray? mappings = null;
+        try
+        {
+            var jsonRoot = JObject.Parse(fileContents);
+
+            if (jsonRoot.TryGetValue("mappings", out JToken? mappingsToken) && mappingsToken?.Type == JTokenType.Array)
+            {
+                mappings = (JArray)mappingsToken;
+            }
+            else
+            {
+                Console.WriteLine($"Cannot find mappings in {SourceMappingsPath}");
+                return [];
+            }
+        }
```

Comment:
We could just parse the file using DarcLib into an existing POCOs with an existing schema to be more type safe. This whole method could be a 2 liner then.